tools/extract_baserom.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: © 2024 ZeldaRET
+# SPDX-License-Identifier: CC0-1.0
+
+from __future__ import annotations
+
+import argparse
+from pathlib import Path
+import sys
+
+import dmadata
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Extract DMA segments from an uncompressed ROM."
+    )
+    parser.add_argument(
+        "rom", metavar="ROM", type=Path, help="Path to uncompressed ROM"
+    )
+    parser.add_argument(
+        "-o",
+        "--output-dir",
+        type=Path,
+        required=True,
+        help="Output directory for segments",
+    )
+    parser.add_argument(
+        "--dma-start",
+        type=lambda s: int(s, 16),
+        required=True,
+        help=(
+            "The dmadata location in the rom, as a hexadecimal offset (e.g. 0x12f70)"
+        ),
+    )
+    parser.add_argument(
+        "--dma-names",
+        type=Path,
+        required=True,
+        help="Path to file containing names of DMA segments",
+    )
+
+    args = parser.parse_args()
+
+    rom_data = args.rom.read_bytes()
```

Comment:
```suggestion
    rom_data = memoryview(args.rom.read_bytes())
```
so that slices won't make a copy, instead will create views of the underlying data
this is a free perf boost